numba/tests/test_parfors.py

```diff
@@ -2495,6 +2495,28 @@ def test_impl(x):
         x = [np.array([1,2,3], dtype=int),np.array([1,2], dtype=int)]
         self.prange_tester(test_impl, x)
 
+    @skip_parfors_unsupported
+    def test_ssa_false_reduction(self):
+        # issue5698
+        # SSA for h creates assignments to h that make it look like a
+        # reduction variable except that it lacks an associated
+        # reduction operator.  Test here that h is excluded as a
+        # reduction variable.
+        def test_impl(image, a, b):
+            empty = np.zeros(image.shape)
+            for i in range(image.shape[0]):
+                r = image[i][0] / 255.0
+                if b == 0:
+                    h = 0
+                if a == r and b != 0:
+                    h = 0
```

Comment:
I'm unsure that this is valid, `h` is not guaranteed to be live and I think this is relying on the zero-init side effect? Does a piece of code not hitting the zero-init suffer the same way?

Reply:
Just realised the PR text has the same concern, apologies, I seem to have missed that!

Reply:
I will put h = 1 at the beginning and then the h=0 in the if's.

Reply:
I think the problem with doing this is that it won't trigger the originally reported problem? Question is... is the patch herein required as a result of cases like the original one whereby zero-init is relied upon or can it also be triggered via some other more legal means?

Reply:
You are right.  I just tested it.  h=1 outside the prange triggers the original problem but h=1 inside the prange does not.  Suggestions?

Reply:
I'm wondering if the analysis in parfors is in fact correct as-is, but being broken by this zero init. In which case practically we have to fix it with respect to the zero-init but probably note in the source that that's what it's doing. I think that the original sample code, whilst technically invalid, is likely to exist in practice and so has to be handled. Unless you can think of a way of triggering this problem from more "legal" code, I suggest:

1. Switch the sample back to the original form with a missing `h` initialiser.
2. Adding to the note here: https://github.com/numba/numba/pull/5705/files#diff-054901b0779b9ed9f103e763d25f872fR3318-R3320 explaining likely cause and why it's needed.

Hows that sound?